scripts/enterenv

```diff
@@ -1,40 +1,75 @@
 #!/bin/bash
 
-IFS="+" read USER ENV <<< "${USER:?}"
-ENV="${ENV:-$USER}"
+# A temporary varible to help checkin whether the environment has been explicitly
+# set to blank
+ENVIRONMENT_SET='0'
+if [[ "${USER:?}" =~ '+' ]]; then
+  ENVIRONMENT_SET='1'
+fi
+
+IFS="+" read USER ENV <<< "${USER}"
+
+THROWAWAY='0'
+if [ "$ENVIRONMENT_SET" = '1' ] && [ -z "$ENV" ]; then
+  THROWAWAY='1'
+else
+  ENV=$USER
+fi
+
+# There's no need for this to be around anymore from this point on
+unset ENVIRONMENT_SET
+
+# The path relative to the host Docker machine where data will be persisted, used
+# for adjusting paths when nesting containers
 HOST_DATA="${HOST_DATA:?}"
 
+# This is where users environments will be stored
+USER_ROOT="${USER_ROOT:-"/data/users/$USER"}"
+
+# Not really needed to be defined as a variable but allows for injecting its value
+# from outside (might be useful for testing purposes)
+GH_ENVY_HOME=${GH_ENVY_HOME:-"https://github.com/$USER/envy.home.git"}
+
+# The obvious ones
+ENV_IMAGE="$USER/$ENV"
+ENV_ROOT="$USER_ROOT/envs/$ENV"
+DIND_CONTAINER_NAME="$USER.$ENV"
+
 data-init() {
-  mkdir -p /data/users
+  mkdir -p $(readlink -f "${USER_ROOT}/..")
   cp /bin/envcmd /data/.envcmd
 }
 
 user-init() {
-  if [[ ! -d "/data/users/$USER" ]]; then
-    mkdir -p "/data/users/$USER"
-    mkdir -p "/data/users/$USER/envs"
-    mkdir -p "/data/users/$USER/home"
-    cp -R /tmp/data/root "/data/users/$USER/root"
+  if [[ ! -d "$USER_ROOT" ]]; then
+    mkdir -p "$USER_ROOT"
+    mkdir -p "$USER_ROOT/envs"
+    if $(wget --spider $GH_ENVY_HOME &>/dev/null); then
+      git clone $GH_ENVY_HOME "$USER_ROOT/home"
```

Comment:
Cause we have the tools for a plugin model, but I just want to wait until we have clear ideas for use. Dokku got out of control really fast. Good and bad.

Anyway, seems like you could do this from within the container via startup scripts. Esp once we have the /root startup script load your user startup script.